client/src/main/java/io/pravega/client/stream/SegmentReader.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pravega.client.stream;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Allows to read data from a specific segment.
+ */
+public interface SegmentReader<T> extends AutoCloseable {
+
+    /**
+     * Gets the next event in the segment. If there are no events currently available this will block up for
+     * timeoutMillis waiting for them to arrive. If none do, an EventReadWithStatus {@link EventReadWithStatus} will be returned with null for
+     * {@link EventReadWithStatus#getEvent()}.
```

Comment:
Can we also add the status as  AVAILABLE_LATER along with event as null when no events in segment